### PR TITLE
No more throwing away janiborg garbage bags

### LINKED ISSUE #5275: Cyborg janitors can drop their trash bag down disposals
As per title - an unfortunate cargo tech got their hands glued, and it also triggered an infinite loop.

### PR DESCRIPTION
**Pull request by Crazylemon64**
Also, engineering borgs/drones can now disassemble disposal bins
Fixes #5275

:cl:Crazylemon
fix: Janiborg trash bags no longer fit down disposals when empty
/:cl:

### CODE REVIEW

File: code/modules/recycling/disposal.dm [hunk 1]
```diff
@@ -143,6 +143,8 @@
 		return
 
 	if(!I)	return
+	if(I.flags & NODROP)
+		return
 
 	user.drop_item()
```
Comment (Fox-McCloud) [inline code/modules/recycling/disposal.dm]: I'm pretty sure that you can get rid of the above check and just have
```
if(!user_drop_item()
     return
```

@TheDZD knows a bit more about the proper unequip usage, especially in relation to NODROP though.

File: code/modules/space_management/TODO [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# Give credit to https://github.com/tgstation/tgstation/pull/12370 for the
+# map transition update
+
+# Tie various systems into the z level manager
+## Radio
+## Gravity
+## Weather
+// TODO: Tie into space manager
+// TODO: Make the radio system cooperate with the space manager
+// TODO: Make the gravity generator cooperate with the space manager
```
Comment (Fox-McCloud) [inline code/modules/space_management/TODO]: uhhhh

File: code/modules/recycling/disposal.dm (hunk 1 above)
Comment (TheDZD) [inline code/modules/recycling/disposal.dm]: This is correct.

**Comment by FreeStylaLT** [comment]
To my knowledge _no one_ can disassemble the damn bins..

File: code/modules/space_management/TODO (hunk 2 above)
Comment (Crazylemon64) [inline code/modules/space_management/TODO]: Aaagh consarn it

**Comment by Crazylemon64** [comment]
Screwdriver, welder, wrench @FreeStylaLT

**Comment by Fox-McCloud** [comment]
@Crazylemon64 there's a few instances where it's unremoveable though. If the drunk underneath a bin is destroyed, you're screwed and need admin help to delete it.

**Pull request merged**